src/Bicep.LangServer/Completions/BicepCompletionProvider.cs

```diff
@@ -69,62 +69,32 @@ private IEnumerable<CompletionItem> GetDeclarationCompletions(BicepCompletionCon
-                yield return CreateContextualSnippetCompletion(LanguageConstants.VariableKeyword, "Variable declaration", "var ${1:Identifier} = $0", context.ReplacementRange);
 
                 yield return CreateKeywordCompletion(LanguageConstants.ResourceKeyword, "Resource keyword", context.ReplacementRange, priority: CompletionPriority.High);
-                yield return CreateContextualSnippetCompletion(LanguageConstants.ResourceKeyword, "Resource with defaults", @"resource ${1:Identifier} 'Microsoft.${2:Provider}/${3:Type}@${4:Version}' = {
-  name: $5
-  location: $6
-  properties: {
-    $0
-  }
-}", context.ReplacementRange, priority: CompletionPriority.High);
-                yield return CreateContextualSnippetCompletion(LanguageConstants.ResourceKeyword, "Child Resource with defaults", @"resource ${1:Identifier} 'Microsoft.${2:Provider}/${3:ParentType}/${4:ChildType}@${5:Version}' = {
-  name: $6
-  properties: {
-    $0
-  }
-}", context.ReplacementRange, priority: CompletionPriority.High);
-                yield return CreateContextualSnippetCompletion(LanguageConstants.ResourceKeyword, "Resource without defaults", @"resource ${1:Identifier} 'Microsoft.${2:Provider}/${3:Type}@${4:Version}' = {
-  name: $5
-  $0
-}
-", context.ReplacementRange, priority: CompletionPriority.High);
-                yield return CreateContextualSnippetCompletion(LanguageConstants.ResourceKeyword, "Child Resource without defaults", @"resource ${1:Identifier} 'Microsoft.${2:Provider}/${3:ParentType}/${4:ChildType}@${5:Version}' = {
-  name: $6
-  $0
-}", context.ReplacementRange, priority: CompletionPriority.High);
 
                 yield return CreateKeywordCompletion(LanguageConstants.OutputKeyword, "Output keyword", context.ReplacementRange);
-                yield return CreateContextualSnippetCompletion(LanguageConstants.OutputKeyword, "Output declaration", "output ${1:Identifier} ${2:Type} = $0", context.ReplacementRange);
 
                 yield return CreateKeywordCompletion(LanguageConstants.ModuleKeyword, "Module keyword", context.ReplacementRange);
-                yield return CreateContextualSnippetCompletion(LanguageConstants.ModuleKeyword, "Module declaration", @"module ${1:Identifier} '${2:Path}' = {
-  name: $3
-  $0
-}", context.ReplacementRange);
 
                 yield return CreateKeywordCompletion(LanguageConstants.TargetScopeKeyword, "Target Scope keyword", context.ReplacementRange);
 
-                foreach (ResourceSnippet resourceSnippet in ResourceSnippetsProvider.GetResourceSnippets())
+                foreach (Snippet resourceSnippet in SnippetsProvider.GetTopLevelNamedDeclarationSnippets())
                 {
-                    yield return CreateContextualSnippetCompletion(resourceSnippet.Name,
+                    string label = resourceSnippet.Prefix;
+                    CompletionPriority completionPriority = CompletionPriority.Medium;
+
+                    if (label.StartsWith("resource"))
```

Comment:
>if (label.StartsWith("resource")) [](start = 20, length = 33)

Can you move this check into the snippet provider and expose the priority or similar property on the snippet object? This way this naming convention is encapsulated inside the provider and the completion provider doesn't have to know about it.

Reply:
Sure, updated here:
https://github.com/Azure/bicep/commit/6827ad852077e54547166090f1aa5a7b0be74da1